### ISSUE TITLE
[Bug] [CLI] Argparse can't handle providers where provider #1 field = 'str' and provider #2 field = choices=[CHOICES].

### ISSUE THREAD

**Issue by deeleeramone** [OpenBB-finance/OpenBB#6573]
**Describe the bug**
^

Constrained choices in one provider impacts the use of all other providers.

**To Reproduce**
```
/derivatives/futures/curve --symbol CL --provider yfinance
```

**Screenshots**

![Screenshot 2024-07-09 at 7 39 19 PM](https://github.com/OpenBB-finance/OpenBB/assets/85772166/84ec6447-1580-4f88-b56d-2e26df259ddd)

**Comment by hjoaquim** [comment]
This might be an edge case and need to look into how a standard required argument deals with provider specific argument with choices - we've solved a similar issue in the past but for non required arguments.

Another thing is, does it make sense for `cboe` to only have those two symbols?
![image](https://github.com/user-attachments/assets/4bb3aef3-24ea-4916-adc0-99c666944792)

**Comment by deeleeramone** [comment]
> Another thing is, does it make sense for `cboe` to only have those two symbols?

Yes, this is explicitly defined because those are the only symbols that can be returned by the provider.
